python/cudf_polars/tests/test_mapfunction.py

```diff
@@ -93,3 +93,14 @@ def test_unpivot_defaults():
     )
     q = df.unpivot(index="d")
     assert_gpu_result_equal(q)
+
+
+def test_with_row_index_defaults():
+    lf = pl.LazyFrame(
+        {
+            "a": [1, 3, 5],
+            "b": [2, 4, 6],
+        }
+    )
+    q = lf.with_row_index()
+    assert_ir_translation_raises(q, NotImplementedError)
```

Comment:
Is this easy enough to implement? Or do we need rust changes too?

Reply:
Yeah this should be easy to implement. Let me try it